webapp/src/ts/modules/contacts/contacts.component.ts

```diff
@@ -428,22 +451,166 @@ export class ContactsComponent implements OnInit, OnDestroy{
     return contact._id + contact._rev;
   }
 
-  // const setActionBarData = () => {
-  //   ctrl.setLeftActionBar({
-  //     hasResults: ctrl.hasContacts,
-  //     userFacilityId: usersHomePlace && usersHomePlace._id,
-  //     childPlaces: allowedChildPlaces,
-  //     exportFn: function() {
-  //       Export('contacts', ctrl.filters, { humanReadable: true });
-  //     },
-  //   });
-  // };
-
-  // const updateAllowedChildPlaces = () => {
-  //   XmlForms.listen('ContactsListCtrl',{ contactForms: true }, (err, forms) => {
-  //     const allowCreateLink = contactType => forms && forms.find(form => form._id === contactType.create_form);
-  //     allowedChildPlaces = childPlaces.filter(allowCreateLink);
-  //     setActionBarData();
-  //   });
-  // };
+  private setLeftActionBar() {
+    this.globalActions.setLeftActionBar({
+      hasResults: this.hasContacts,
+      userFacilityId: this.usersHomePlace && this.usersHomePlace._id,
+      childPlaces: this.allowedChildPlaces?.sort((a, b) => a.id.localeCompare(b.id)),
+      exportFn: () => {
+        this.exportService.export('contacts', this.filters, { humanReadable: true });
+      }
+    });
+  }
+
+  private subscribeToSelectedContact() {
+    const subscription = this.store
+      .select(Selectors.getSelectedContact)
+      .subscribe((selected) => {
+        this.selectedContact = selected;
+        this.setChildTypesBySelectedContact();
+
+        this.globalActions.setRightActionBar({
+          relevantForms: [], // This disables the "New Action" button in action bar until forms load
+          sendTo: this.selectedContact?.type?.person ? this.selectedContact?.doc : '',
+          canDelete: !!this.selectedContact?.children?.every(group => !group.contacts?.length),
+          canEdit: this.sessionService.isAdmin() || this.userSettings?.facility_id === this.selectedContact?.doc?._id,
+          openContactMutedModal: (event, form) => this.openContactMutedModal(form),
+          openSendMessageModal: (sendTo) => this.openSendMessageModal(sendTo)
+        });
+
+        this.subscribeToAllContactXmlForms();
+        this.subscribeToSelectedContactXmlForms();
+      });
+    this.subscription.add(subscription);
+  }
+
+  private openContactMutedModal(form) {
+    if (!form.showUnmuteModal) {
+      this.router.navigate(['/contacts', this.selectedContact._id, 'report', form.code]);
+      return;
+    }
+
+    this.modalService
+      .show(ContactsMutedComponent)
+      .then(() => this.router.navigate(['/contacts', this.selectedContact._id, 'report', form.code]))
+      .catch(() => {});
+  }
+
+  private openSendMessageModal(sendTo) {
+    this.modalService
+      .show(SendMessageComponent, { initialState: { fields: { to: sendTo } } })
+      .catch(() => {});
+  }
+
+  private async setChildTypesBySelectedContact() {
```

Comment:
Please don't mix async with promises in the same file.

Reply:
I think this keep code cleaner - flat, other wise too many levels in callbacks or can lead to over modularisation. 
We have several places with sequential code and we can start moving them to async-await little by little.